Grues Revisited

Hinaichigo opened the pull request:
This adds grues, monsters that lurk in the dark and eat you if you aren't too careful. The idea is to have them as an antag.
They have some light-based mechanics which I think could make for interesting gameplay. They were originally added as humanoid creatures [(/pull/12848)] by Shadowmech88 but I was thinking they could be revisited with different mechanics and aesthetics. The name could be changed if this is seen as a conflict.

Grue features:
- Healed by darkness and burned by light
- Starts out in a weak form and gradually evolves
- Eat people to upgrade your power
- See in the dark with Grue-Vision™

![grue1](https://user-images.githubusercontent.com/7112773/146784818-30e83c24-ad43-4192-9148-33212d7fcbf2.png)
> _A grue eating someone in maint_.


cl
* rscadd: Added a new antagonist: the grue; a creature that lurks in the dark and eats you if you aren't too careful.
* rscadd: Grues thrive in the dark and abhor bright lights. They heal and move faster in the dark and get burned in bright light.
* rscadd: Grues have four life stages: egg, larva, juvenile, and adult.
* rscadd: Adult grues can eat people whole, and doing so makes them more powerful each time. Beware the snowball effect.
* rscadd: Grues are particularly vulnerable while moulting into a new life stage.
* rscadd: Egg laying can be toggled as a config option.
* rscadd: Meat can be recolored without having to add custom meat sprites for each mob.
* bugfix: Slime puddles can now be put into cloners and scanned properly.
* tweak: Renamed humanoid grue to umbra.

Eneocho commented:
Are grues no longer snowflakey human mobs? Because as they currently are they're just recoloured humans with race dependent gene powers.

Hinaichigo commented:
> Are grues no longer snowflakey human mobs? Because as they currently are they're just recoloured humans with race dependent gene powers.

They're now more monstrous and start out as larva and then have a juvenile stage and an adult stage. The adult stage can eat people and then lay eggs that hatch into larva.

Eneocho commented:
So more like a benomorph? How many people does a single grue need to eat to max their powers? How does the "people eating" work? Is it completely consuming the mob, like the spiders that suck a bit of juice, like the benos that cocoon and or like the slimes that "eat" you?

I feel this will have similar round ramifications as a benoloose.

Eneocho commented:
By the way, you can tag this PR as a draft is you don't want it merged yet, there should be a link next to the PR name that says "mark as draft" or similar.

Hinaichigo commented:
> So more like a benomorph? How many people does a single grue need to eat to max their powers? How does the "people eating" work? Is it completely consuming the mob, like the spiders that suck a bit of juice, like the benos that cocoon, like the zombies that can bite a few times to "feed" or like the slimes that gene fuck you?
> 
> I feel grues will have similar round ramifications as benos.

Eating people works by targeting a mob (specifically a carbon) first, then the grue opens its mouth and prepares to eat for 10 seconds. If the grue moves or the target moves out of range, it cancels. If it succeeds the mob is gibbed. If the mob had a mind  (to prevent farming monkeys) the grue gains some power, which gives incremental stat boosts (increased speed, melee damage, health, and health regen in darkness), and the ability to break down normal walls at 3 people eaten and reinforced walls at 4. Though the grue still requires oxygen which prevents it from wanting to vent the atmosphere into space. Also the grue can lay one egg for every person eaten.  I was thinking of giving them the ability to shroud themselves in darkness after eating 2 people but I'm not sure if that'd be good gameplay wise so I left it out for now. The powers and stat boosts are something that I very much want feedback on and might need to be tweaked after in-round testing.

Eneocho commented:
I'd either remove the gibbing (instead husking the body so it needs cloning) or make the gib spawn a head so the guy can be revived with a head transplant.

Hinaichigo commented:
> I'd either remove the gibbing (instead husking the body so it needs cloning) or make the gib spawn a head so the guy can be revived with a head transplant. Nobody likes being round removed and some might want to revive and fight back instead of joining the grue hive.

Makes sense. Maybe a glob of gore with enough DNA that it could be put into the cloner, similar to slime people.

Eneocho commented:
Slime blobs can't be cloned tragically. The only way for people to be cloned is to be in a clonable body (not that much gene damage) with a healthy brain inside. 
Either make the grue eat the body but leave the head, something about not wanting to get mad spess disease or "eat" the body like zombies do.
Head transplant is cumbersome and slow so a fast grue hive could round remove everyone before the WILD RIDE can revive 5 people, but just outright defibbing a corpse is way too easy a fix. Husking the body (something something, human bodies produce infrared light and they eat you to remove said light) is a good middleground, since you can still get revived via cloning but isn't as fast as defibbing or as slow as head transplanting. It would also encourage m*chanics to make spare cloners.

Jeff-jefferson-jeffery commented:
Anyone in line of sight of the grue should get an "IT IS DARK, YOU ARE LIKELY TO BE EATEN BY A GRUE" message.

Eneocho commented:
>Anyone in line of sight of the grue should get an "IT IS DARK, YOU ARE LIKELY TO BE EATEN BY A GRUE" message.

That's not fun for the grue and wouldn't feel organic.

TGFH commented:
Reading Eneocho posts is not fun for anyone.

Hinaichigo commented:
I added cloneable/brain-extractable gore globs when a grue eats you. They are basically slime puddles. Also fixed some bugs preventing slime puddles from being put into the cloner.

Eneocho commented:
> I added cloneable/brain-extractable gore globs when a grue eats you. They are basically slime puddles. Also fixed some bugs preventing slime puddles from being put into the cloner... but I still need to debug it a bit.

Guess that works. Thanks for fixing the slime puddles too.

Hinaichigo commented:
It seems like gibbing already produces a head with a brain, so I just kept it as is (but with the slime puddle fixes). Also polished up few more things here and there, I think I'd like to open this for review again.

GreatBigFailure commented:
If you do finish this please maintain current grues as their own thing. Rename them space demons or whatever. They're a very fun midround race in the rare instance you get to play one.
No extra code needed for the species version, just a rename

Eneocho commented:
Grue, the anti CIA glowie.

Hinaichigo commented:
I can do that sure, what kind of name would be good for the species grues?

Eneocho commented:
> I can do that sure, what kind of name would be good for the species grues?

Tenebris for that latin feel.

Hinaichigo commented:
I renamed the humanoid grues to Umbras, and fixed up a bit of the sound effects. It's essentially ready for now.

Hinaichigo commented:
How would I get the "diffs" check to run again without adding another commit?

Eneocho commented:
Does the airlock forcing have a delay? If it does, consider removing it, being able to open airlocks at will while running away helps a lot. For reference, player controllable spiders don't have a delay either.

Hinaichigo commented:
> Does the airlock forcing have a delay? If it does, consider removing it, being able to open airlocks at will while running away helps a lot. For reference, player controllable spiders don't have a delay either.

I added a delay so that grues couldn't quickly run through a well-lit area and then duck back into maintenance to heal in the dark. It's 12 seconds for juveniles, 10 seconds for adults, and decreases by 4 seconds for every person eaten.
I could remove it but I sort of want to discourage grues from 'tanking' the bright light, (aside from popping out to smash a nearby light and thereby expand the dark area), so that they aren't seen as often out in the open. Another alternative to the door forcing timer is to make them slower in the light (right now they are human speed in the light but speed up slightly in the dark) so they would be bogged down and have a harder time going through the light, or maybe increasing the damage light does. What would be the most interesting and fun? Some of this is hard to tell without testing it in the context of a live round first.

Eneocho commented:
>Maybe if juveniles have a several second delay whereas adults can do it instantaneously?

That works. Any kind of evolving antag has some serious weakness against even a barely mobilized crew, specially during the early stages. It's why playerspiders camp on deep maint then rush to genetics or virology when they're close to evolving (you don't get growthtick while inside pipes) or why benos do the same.
Making it a juvenile only issue would go along with the other evolving antags.

Hinaichigo commented:
Okay, done. I reduced juvenile's to 10 seconds too.

ShiftyRail commented on code/_onclick/ventcrawl.dm:
```diff
@@ -133,6 +133,13 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 	return FALSE
 
+/mob/living/simple_animal/hostile/grue/can_ventcrawl()
+	if(lifestage==1)
```
Please use a define for this.

Hinaichigo replied:
Could you please let me know what you mean by this? I'm not quite sure. lifestage can range from 1 to 3 so I want to be sure that only the larval form (lifestage==1) can ventcrawl.

ShiftyRail replied:
`#define GRUE_LARVAL 1` and then `if (lifestage == GRUE_LARVAL)`. This way anyone can understand which stage it is without remembering that 1 is the larva, etc.

ShiftyRail commented on code/_onclick/ventcrawl.dm:
```diff
@@ -133,6 +133,13 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
+	if(lifestage==1)
+		return TRUE
+	else
+		to_chat(src, "<span class='notice'>You are too big to fit into the pipe.</span>")
```
A `warning` class would fit better here.

Hinaichigo replied:
I wasn't sure but I initially put notice to be consistent with fat mice and alien queens.

ShiftyRail commented on code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm:
```diff
@@ -197,6 +197,66 @@
+/datum/dynamic_ruleset/latejoin/grue
+	name = "Grue Infestation"
+	role_category = /datum/role/grue
+//	enemy_jobs
```
Please use empty lists for this instead of this commented out var.

Hinaichigo replied:
Done.

ShiftyRail commented on code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm:
```diff
@@ -197,6 +197,66 @@
+	return ..()
+
+/datum/dynamic_ruleset/latejoin/grue/execute()
+	var/mob/M = pick(assigned)
```
this looks like you should double check if `makeBody` or some similar is set to 0. You may or may not get bugs similar to grinch if you don't.

Hinaichigo replied:
It seems like makeBody is only defined for midround rulesets, and defaults to 0. I could explicitly set it to 0 in dynamic_rulesets_midround.dm if that's what you mean.

ShiftyRail replied:
That's fine if it's set to 0. I misrembered.

ShiftyRail commented on code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm:
```diff
@@ -197,6 +197,66 @@
+	var/turf/oldloc = get_turf(M)
+	M.forceMove(null)
+	if(!latejoinprompt(M,src))
+		message_admins("[M.key] has opted out of becoming a grue.")
```
I don't like this as it may just waste a ruleset injection. But that's not a deal breaker.

Hinaichigo replied:
What would be a better alternative? I guess just forcing it since they presumably had the role preference  set anyway?

ShiftyRail replied:
This check should be in `ready` or something similar rather than in `execute`. Although I am not sure if a failed execute wastes the ruleset or not. I would have to double check. I'll get back.

ShiftyRail commented on code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm:
```diff
@@ -849,6 +849,54 @@
+/datum/dynamic_ruleset/midround/from_ghosts/grue
+	name = "Grue Infestation"
+	role_category = /datum/role/grue
+//	enemy_jobs
```
same as above

ShiftyRail commented on code/datums/gamemode/role/grue.dm:
```diff
@@ -0,0 +1,31 @@
+	id = GRUE
+	special_role = GRUE
+	required_pref = ROLE_GRUE
+//	wikiroute = ROLE_GRUE
```
don't comment this out. A blank wiki page is better than hiding the problem.

Hinaichigo replied:
Done.

ShiftyRail commented on code/datums/gamemode/role/grue.dm:
```diff
@@ -0,0 +1,31 @@
+
+/datum/role/grue/Greet()
+	to_chat(antag.current, "<span class='danger'>You are a grue.</span>")
+	to_chat(antag.current, "<span class='danger'>Darkness is your ally; bright light is harmful to your kind. You hunger... specifically for sentient beings, but you are still young and cannot eat until you are fully mature.</span>")
```
Change danger to something else here, but that's mostly style. Maybe warning.

Hinaichigo replied:
Done.

ShiftyRail commented on code/datums/gamemode/role/grue.dm:
```diff
@@ -0,0 +1,31 @@
+	. = ..()
+	if(istype(antag.current,/mob/living/simple_animal/hostile/grue))
+		. += "<BR>The grue ate [eatencount] sentient beings."
+//		if(config.grue_egglaying)
```
You don't really need to make this a config setting. Why is this commented out ?

Hinaichigo replied:
It was commented out because I wasn't sure how to actually update these role-defined variables in-game as the grue eats or spawns offspring. Am I correct in thinking it's something like this:
`M.mind.role.eatencount++`
?

ShiftyRail replied:
No. I would be something like this:
```dm
var/datum/role/grue/G = M.mind.GetRole(ROLE_GRUE)
G.eatencount++
```

ShiftyRail commented on code/game/dna/dna_modifier.dm:
```diff
@@ -239,35 +240,35 @@
-		if(user)
-			to_chat(user, "<span class='notice'>For some reason, the scanner is unable to read that person's genes.</span>")
+	if(ismanifested(M) || (!iscarbon(M) && !istype(M, /mob/living/slime_pile) && !istype(M, /mob/living/gore_pile)))
+		if(usr)
```
Never use `usr`. Please use `user` instead. Why did you make this change ?

Hinaichigo replied:
user was undefined and it was a longstanding bug in the cloner code where checks like if(user) would always return false.
usr is the hardcoded thing akin to src as I understand.

ShiftyRail replied:
`usr` is hardcoded to be "the last player to have done a thing (in a context related to `src`)". You have no control over it. It's hard to see the code flow and debug things with `usr`. DM's own reference recommends against using `usr` in procs that are not natively called by BYOND, such as `Topic`, etc.

If `user` is null, then the bug is something further above and that should be fixed instead of relying on `usr`.

ShiftyRail commented on code/modules/mob/living/simple_animal/grue_egg.dm:
```diff
@@ -0,0 +1,131 @@
+	maxHealth=25
+	health=25
+
+
```
tralling spaces (just a style comment, the same applies to your rulesets file which lack a space before the ruleset procs

Hinaichigo replied:
On a related note, what is best-practice for end-of-file? Have a newline or no?

ShiftyRail replied:
No new lines at the end of files. Try to keep tralling lines to a minimum. Try to make embed logical expression readable. That's more or less all there is to it.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	force_airlock_time=100 									//so that juvenile grues cant easily rush through a light area and quickly force open a door to escape back into the dark
+
+	//VARS
+//	var/isgrue=1
```
Remove this comment if it's not used.

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	var/a_53 = 0
+	var/a_54 = 0
+
+//	var/b_matrix_testing_override=0 //not used for now
```
ditto

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	var/a_use_alpha=1
+	var/a_blend_add_test=255
+
+	var/a_matrix_testing_override = 0 //not used for now
```
What is this for ?

Hinaichigo replied:
For testing and fine-tuning the color matrix to get grue-vision working. I pulled it from the spider nightvision code.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+			current_brightness=10*T.get_lumcount()
+		else												//else, there's considered to be no light
+			current_brightness=0
+//		visible_message("<span class='warning'>\The [src] is in brightness level [current_brightness] with [health] health and [shadowpower] shadowpower.</span>") //debug
```
No dead comments. Maybe make this proc every 300 life ticks (there is a var for that) if spam is an issue.
```dm
if ( (lifetick % 300) == 0)
    ...
```

Hinaichigo replied:
Sorry, that was just for debugging.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+
+/mob/living/simple_animal/hostile/grue/proc/lifestage_updates() //Initialize or update lifestage-dependent stats
+	var/tempHealth=health/maxHealth
+	if(lifestage==1)
```
defines

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+/mob/living/simple_animal/hostile/grue/proc/lifestage_updates() //Initialize or update lifestage-dependent stats
+	var/tempHealth=health/maxHealth
+	if(lifestage==1)
+		name = "grue larva"
```
maybe make this mob subtypes

Hinaichigo replied:
The reason I did it like this is to make it easier to switch between life stages as the grue moults and such. Do you mean it would be better to spawn a new mob, transfer the mind in, and then delete the original mob?

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		attack_sound = 'sound/weapons/bite.ogg'
+		size = SIZE_SMALL
+		pass_flags = PASSTABLE
+	else if (lifestage==2)
```
ditto

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+				stat(null, "Reproductive energy: [eatencharge]")
+			stat(null, "Sentient life forms eaten: [eatencount]")
+
+/mob/living/simple_animal/hostile/grue/gruespawn
```
Remove the var overrides you did previously and put them here.

Hinaichigo replied:
I'm not sure what you mean by this. Could you please clarify?

Hinaichigo replied:
The variables change as the grue moults into a new lifestage, so by having them changed by the lifestage_updates() proc this lets the variables be set by both New() and by moulting. If I changed it from that, then they wouldn't update properly after moulting.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		to_chat(src, "<span class='notice'>You are already fully mature.</span>")
+
+/mob/living/simple_animal/hostile/grue/proc/start_moult()
+	if(stat==CONSCIOUS&&shadowpower>=moultcost&&!ismoulting&&lifestage<3)
```
Style comment : space between operators (at leat for && and ==)

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+			icon_dead = "moult2"
+			maxHealth=50 //vulnerable while moulting
+		health=tempHealth*maxHealth //keep same health percentage
+	else
```
useless, remove

Hinaichigo replied:
This is so that a grue at 50 out of 100 (50%) health will turn into a chrysalis at 25 out of 50 (50%) health instead of it becoming a full health chrysalis.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		else if(lifestage==3)
+			to_chat(src, "<span class='warning'>You finish moulting! You are now fully-grown, and can eat sentient beings to gain their strength.</span>")
+		playsound(src, 'sound/effects/lingextends.ogg', 50, 1)
+	else
```
useless, remove

Hinaichigo replied:
Remove the sound effect? I figured it could add a little bit of aesthetic flavor.
Or do you mean the else return?

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	..()
+
+/mob/living/simple_animal/hostile/grue/attack_animal(mob/living/simple_animal/M)
+	if(M==src) //Prevent the grue from attacking itself, might help avoid misclicks while attempting to smash lights.
```
don't like this

Hinaichigo replied:
Upon further reflection yeah it's not good; removed.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		return
+	else
+//		if(prob(20)&&lifestage>1)
+//			playsound(src, 'sound/misc/grue_growl.ogg', 50, 1) //occasionally growl while attacking
```
Why is this commented ?

Hinaichigo replied:
I commented it out in order to reserve that sound effect for a successful feeding. Should it go back in?

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+
+
+//Reproduction via egglaying.
+/mob/living/simple_animal/hostile/grue/verb/reproduce()
```
Ideally you would want this to be a spell, but this is going the extra mile.

Hinaichigo replied:
I'll look into the spells. It could also help hide all the lifestage-specific verbs when they're not usable, if I understand correctly.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	set name = "Reproduce"
+	set desc = "Spawn offspring in the form of an egg."
+	set category = "Grue"
+	if (!config.grue_egglaying) //Check if egglaying is enabled.
```
No real need for a config flag....

Hinaichigo replied:
I was thinking, just in case people don't like how the rounds play out with egglaying enabled.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		busy=0
+
+	else
+		to_chat(src, "You need to feed more first.")
```
Maybe wrap this is span classes ?

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	if(eatencharge>=1)
+		busy=1
+		to_chat(src, "<span class='notice'>You start to push out an egg...</span>")
+		visible_message("<span class='warning'>[src] tightens up...</span>")
```
`\The [src]`

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		visible_message("<span class='warning'>[src] tightens up...</span>")
+		if(do_after(src, src, 5 SECONDS))
+			to_chat(src, "<span class='notice'>You lay an egg.</span>")
+			visible_message("<span class='warning'>[src] pushes out an egg!</span>")
```
`\The [src]`

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	var/list/candidates=list()
+	for(var/mob/dead/observer/G in get_active_candidates(ROLE_GRUE, poll="Would you like to become a grue?"))
+		if(!G.client)
+			//testing("Client of [G] inexistent")
```
Remove these comments.

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+
+	return 0
+
+/mob/living/simple_animal/hostile/grue/proc/transfer_personality(var/client/candidate)
```
I am not sure if this is needed. Maybe the antag role `post_mind_transfer` is a better place for this.

Hinaichigo replied:
I'll check that out.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+	shuffle(candidates)
+	for(var/mob/i in candidates)
+		if(!i || !i.client)
+			continue //Dont bother removing them from the list since we only grab one wizard
```
A very dark wizard, then.

Hinaichigo replied:
Heh, was like that when I found it.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+
+
+//Eating sentient beings.
+/mob/living/simple_animal/hostile/grue/verb/eat_sentient()
```
This one definitively should be a spell. Ask people to help you sprite it if you need.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+			force_airlock_time=max(0,force_airlock_time-40)
+//			src.set_light(8,-1*eatencount) //gains shadow aura opon eating someone
+			switch(eatencount)
+				if(3)
```
use defines for this.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue.dm:
```diff
@@ -0,0 +1,591 @@
+		return
+
+//Ventcrawling and hiding, only for gruespawn
+/mob/living/simple_animal/hostile/grue/verb/ventcrawl()
```
Ideally, this should be, again, a spell.

Hinaichigo replied:
I'll see if I can move everything to spells.

ShiftyRail commented on code/modules/mob/living/simple_animal/hostile/grue_egg.dm:
```diff
@@ -0,0 +1,132 @@
+
+mob/living/simple_animal/grue_egg
```
Style comment : `/mob/...`

Hinaichigo replied:
Done.

ShiftyRail commented on code/modules/organs/organ.dm:
```diff
@@ -77,10 +77,10 @@
-/mob/living/carbon/human/var/list/datum/organ/external/organs_by_name = list() //Map organ names to organs
-/mob/living/carbon/human/var/list/datum/organ/internal/internal_organs_by_name = list() //So internal organs have less ickiness too
-/mob/living/carbon/human/var/list/grasp_organs = list()
+/mob/living/var/list/organs = list()
```
Not sure if this is warranted. Could have unintended consequences.

Hinaichigo replied:
It was done to get slime puddles to fit into the DNA scanner. But yeah it's not ideal. I'll see if I can find another way.

The cloner has a check to see if the subject's brain's owner's DNA is different from the subject itself's DNA, and prioritizes the former for the clone body if so. So to not do this change to organs and give an empty list for all living mobs, we'd have to either remove that check (and thus ignore any unusual cases where the subject's brain's owner's DNA differs from the subject's DNA) or make two separate cloning processes, one for humans and one for slime puddles, etc.
Are there any situations where a subject in a cloner would have a brain owner with DNA different from the DNA of the subect itself? The organ transplant code automatically updates the new organ owner, so I don't think so, so I'll go ahead and revert this and then remove that check from cloning.dm, but please let me know if there is something I'm missing.

ShiftyRail commented on html/changelogs/Hinaichigo.yml:
```diff
@@ -0,0 +1,11 @@
+author: Hinaichigo
```
You don't need a .yml changelog. Rather, put it in the body of your PR.

ShiftyRail commented:
Apologies for the late review. Please take a look at it. Using spells would be very good, as currently all our antags use spells for special abilities, and they are much better than verbs.

Hinaichigo commented:
> Apologies for the late review. Please take a look at it. Using spells would be very good, as currently all our antags use spells for special abilities, and they are much better than verbs.

Thanks for the feedback. I'll look into getting the abilities in the form of spells. Also could you please let me know what you mean by "using defines" instead of checking the value of the lifestage variable?

Hinaichigo commented:
I switched all the abilities over to spells.

Hinaichigo commented:
Does anyone know why the compile and run test might be failing here?
On my local version it compiles and works fine but it fails here for some reason, and I'm not sure why.

Hinaichigo commented:
@ShiftyRail I incorporated many of the changes you recommended. There is still a little bit left, mainly UI and aesthetic things to change, but I was wondering if you could please take another look just to make sure things are on the right track, thanks!

Hinaichigo commented:
I'd like to open this for review again.

Kurfursten commented:
Confl

Hinaichigo commented:
@ShiftyRail I fixed the conflict.

ShiftyRail commented on code/_onclick/hud/other_mobs.dm:
```diff
@@ -300,4 +300,27 @@
+	mymob.healths2.screen_loc=ui_under_health
+
+
+	//vampire_blood_display = new /obj/abstract/screen
```
better just remove this commented block.

ShiftyRail commented on code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm:
```diff
@@ -197,6 +197,67 @@
+
+	repeatable = TRUE
+	var/grue_spawn_spots=list()
+	var/turf/T
```
I am not sure about single-letter vars in datum objects. Within procs it's almost canon, but here it's less excusable.

ShiftyRail reviewed (commented):
that's all chief

Hinaichigo commented:
Made those fixes.

ShiftyRail commented:
i for one am ready for #gruelgate
